provider/equinix/environ.go

```diff
@@ -617,12 +619,23 @@ nextPlan:
 
 func validPlan(plan packngo.Plan, region string) bool {
 	// some plans may not be servers
-	if plan.Pricing == nil ||
+	if plan.Line != "baremetal" ||
+		plan.Pricing == nil ||
 		plan.Specs == nil ||
 		plan.Specs.Memory == nil ||
 		len(plan.Specs.Cpus) == 0 || plan.Specs.Cpus[0].Count == 0 {
 		return false
 	}
+	var validDeploymentType bool
+	for _, d := range plan.DeploymentTypes {
+		if d == "on-demand" {
```

Comment:
You are checking for `on-demand` with a dash here. [Here](https://github.com/juju/juju/pull/13227/files#diff-8bb58109c31e5e10245b3fed89271e67ef0796d716d718a9268a9b42324653e9R572) you provide `on_demand` with underscore and in your test case, some of the plans use the underscore and others use the dash.

Which one is correct?

Reply:
Great catch! `on_demand` is correct:

```
- class: c3.medium.x86
  deployment_types:
  - on_demand
  - spot_market
  description: A killer general-purpose compute box with 24 high-performance AMD EPYC
    cores and 64GB RAM
  id: 359ba4aa-6353-4480-bce2-a6dcd5a69aaa
  legacy: true
  line: baremetal
  name: c3.medium.x86
  pricing:
    hour: 1.1
  slug: c3.medium.x86
  specs:
    cpus:
    - count: 1
      type: AMD EPYC 7402P 24-Core Processor @ 2.8GHz
    drives:
    - count: 2
      size: 240GB
      type: SSD
    - count: 2
      size: 480GB
      type: SSD
    features:
      raid: true
      txt: true
    memory:
      total: 64GB
  ...
```